packages/appsync-modelgen-plugin/src/utils/process-conversation.ts

```diff
@@ -0,0 +1,198 @@
+import { CodeGenConnectionType, Field, FieldAttribute, FieldType, PrimaryKeyInfo, SchemaConversationRoute, SchemaModel, SchemaMutation, SchemaSubscription } from '../interfaces/introspection';
+import { pascalCase } from 'change-case';
+import { plural } from 'pluralize';
+import { CodeGenMutation } from '../visitors/appsync-visitor';
+
+/**
+ * Generates metadata for a conversation model based on a mutation object.
+ * @param mutationObj - The mutation object to generate conversation metadata from.
+ * @param generateGraphQLOperationMetadata - A function to generate GraphQL operation metadata.
+ * @returns The generated conversation route schema.
+ */
+export function processConversationRoute(
+  mutationObj: CodeGenMutation,
+  generateGraphQLOperationMetadata: <T extends CodeGenMutation, V extends SchemaMutation>(operationObj: T) => V
+): SchemaConversationRoute {
+  const routeName = pascalCase(mutationObj.name);
+  const conversationModelName = `Conversation${routeName}`;
+  const conversationMessageModelName = `ConversationMessage${routeName}`;
+
+  return {
+    name: mutationObj.name,
+    models: {
+      [conversationModelName]: generateConversationModel(conversationModelName, conversationMessageModelName),
+      [conversationMessageModelName]: generateConversationMessageModel(conversationModelName, conversationMessageModelName),
+    },
+    nonModels: {},
+    enums: {
+      ConversationParticipantRole: {
+        name: 'ConversationParticipantRole',
+        values: ['user', 'assistant'],
+      }
+    },
+    conversation: {
+      modelName: conversationModelName
+    },
+    message: {
+      modelName: conversationMessageModelName,
+      send: {
+        ...generateGraphQLOperationMetadata<CodeGenMutation, SchemaMutation>(mutationObj),
+        type: { model: conversationMessageModelName }
+      },
+      subscribe: generateSubscriptionMetadata(routeName, conversationMessageModelName),
+    }
+  };
+}
+
+/**
+ * Generates a conversation model schema.
```

Comment:
As discussed, let's add a note here that this duplicates the conversation model creation logic in the transformer, and the two should be evolved in parallel. (Future PR is fine)